python_modules/libraries/dagster-azure/dagster_azure/adls2/fake_adls2_resource.py

```diff
@@ -97,19 +132,20 @@ def delete_file(self, file):
 class FakeADLS2FileClient:
     """Stateful mock of an ADLS2 file client for testing."""
 
-    def __init__(self):
+    def __init__(self, name, fs_client):
+        self.name = name
         self.contents = None
         self.lease = None
+        self.fs_client = fs_client
 
     def get_file_properties(self):
         if self.contents is None:
             raise ResourceNotFoundError("File does not exist!")
-        return {"lease": self.lease}
+        return {"lease": self.lease.id}
```

Comment:
is this potentially a breaking change? Or did this never work to begin with?

Reply:
I was thinking of scenarios where people were previously expecting `self.lease` to be the ID itself, and now the behavior will be slightly changing. Would we expect a user to be interacting with the lease property directly?

Reply:
hmmm I dont think we would be expecting that, but it's not 100% out of the question. I could update to using `self._lease` and add a `lease` property that returns `self._lease.id`